serverless/src/app.js

```diff
@@ -45,36 +46,28 @@ export default async (event) => {
     delete body.STELLAR_NETWORK
     delete body.txFunction
 
-    // const txFunction = new Function(`
-    //   __filename,
-    //   __dirname,
-    //   require,
-    //   module,
-    //   HORIZON_URL,
-    //   STELLAR_NETWORK,
-    //   global,
-    //   globalThis,
-    //   process
-    // `, // leave [global, globalThis, process] empty to effectively unset them
-    // `
-    //   'use strict'; 
-    //   ${txFunctionCode}; 
-    //   return module.exports;
-    // `)(
-    //   __filename,
-    //   __dirname,
-    //   require,
-    //   {exports: null},
-    //   HORIZON_URL,
-    //   STELLAR_NETWORK,
-    // )
-    // const result = await txFunction(body)
+    const vm = new NodeVM({
+      console: 'off',
+      eval: false,
+      wasm: false,
+      strict: true,
+      fixAsync: true,
+      sandbox: {
+        HORIZON_URL,
+        STELLAR_NETWORK
+      },
+      require: {
+        builtin: null,
+        external: ['bignumber.js', 'node-fetch', 'stellar-sdk', 'lodash'],
+        context: 'host'
+      }
+    })
 
-    const result = await eval(`
+    // TODO: handle cases where the module.exports function returned is not a promise
+    const result = await vm.run(`
       'use strict'; 
       ${txFunctionCode};
-      module.exports;
-    `)(body)
+    `, 'vm.js')(body)
```

Comment:
what does `(body)` do?

also, will this error out out if the file `vm.js` doesn't exist?

Reply:
the vm compiles the code and builds a function then `(body)` runs the function with the body as the argument. Exactly the same as what eval did with the exception we don't need to explicitly return module.exports as the vm will do that already as it's meant to compile vs just run and return the result.